Tests: Storage. Покрытие API модуля тестами (добавлены тесты `NFS`)

dr-wh0 opened the pull request:
# Описание изменений

К **интеграционным тестам** API модуля хранилищ (_**Storage**_) добавлены тесты для работы с **NFS**-хранилищами: создание, удаление и получение информации о хранилищах.  
Добавлены специализированные фикстуры для подготовки NFS-хранилищ, томов и шаблонов.

⚠️ Для запуска тестов требуется **настроенный NFS сервер** (обычно на соседней ВМ). В тестовую конфигурацию (`.env.test`) добавлены параметры подключения к NFS (IP, path).

**ВАЖНО: Рекомендуется вливать после _merge_ основного PR: #267** 

# Ключевые изменения

**Конфигурация тестовой среды:**
- Добавлены следующие переменные окружения, указываемые в конфигурации `.env.test` перед тестами:
```
	TEST_STORAGE_NFS_IP=192.168.0.1  # IP-адрес заранее настроенного NFS-сервера
	TEST_STORAGE_NFS_PATH=/var/tmp/nfs_share  # Путь к настроенному каталогу NFS-сервера
```
- В `StorageSettings` добавлены соответствующие поля `storage_nfs_ip` и `storage_nfs_path`.

**Тестовая инфраструктура (фикстуры в `conftest.py`):**
- `storage` — создаёт хранилище и удаляет после теста с параметризацией поля `storage_type` (для тестов NFS - указывается параметр `nfs`)

**Реализованные тесты:**

*Создание (NFS):*
- `test_create_storage_nfs_success` — успешное создание NFS-хранилища
- `test_create_storage_nfs_invalid_ip` — некорректный IP
- `test_create_storage_nfs_invalid_mount_version` — некорректная версия монтирования
- `test_create_storage_nfs_invalid_path` — некорректный путь
- `test_create_storage_nfs_missing_required_fields` — отсутствие обязательных полей (IP/path)
- `test_create_storage_nfs_duplicate_name` — дубликат имени
- `test_create_storage_nfs_unauthorized` —  доступ без авторизации

*Удаление (NFS):*
- `test_delete_storage_nfs_success` — успешное удаление
- `test_delete_storage_nfs_invalid_uuid` — удаление с некорректным UUID
- `test_delete_storage_nfs_not_found` — удаление несуществующего хранилища
- `test_delete_storage_nfs_with_attached_volume` — удаление хранилища с прикреплёнными томами
- `test_delete_storage_nfs_with_attached_template` — удаление хранилища с прикреплёнными шаблонами
- `test_delete_storage_nfs_unauthorized` —  доступ без авторизации

*Получение (NFS):*
- `test_get_storages_nfs_success` — получение списка с NFS-хранилищем
- `test_get_storage_nfs_success` — получение конкретного NFS-хранилища
- `test_get_storage_nfs_nonexistent` — запрос несуществующего
- `test_get_storage_nfs_invalid_id` — запрос с некорректным ID
- `test_get_storage_nfs_unauthorized` — доступ без авторизации

**Domain-изменения:**
- В класс `NfsVolume` добавлен метод `clone()` (аналогичный методу LocalFS), что исправляет issue #269

# Общая структура файлов для тестов

```
tests/
├── conftest.py  # Локальные фикстуры для тестов
├── api/
│    └── localfs/
│         ├── test_create.py  # POST /storages/create/, /storages/local-disks/create_partition/
│         ├── test_delete.py  # DELETE /storages/{id}/delete/, /storages/local-disks/delete_partition/
│         └── test_retrive.py  # GET /storages/, /storages/{id}/, /storages/local-disks/
│    └── nfs/
│        ├── test_create_nfs.py  # POST /storages/create/
│        ├── test_delete_nfs.py  # DELETE /storages/{id}/delete/
│        └── test_retrive_nfs.py  # GET /storages/, /storages/{id}/
```



# Особенности

- На данный момент тесты требуют наличия настроенного NFS-сервера для тестирования NFS-функциональности:

В качестве NFS-сервера для тестов рекомендуется настроить вторую виртуальную машину, связанную с тестируемой ВМ сетевым мостом. Краткая инструкция по настройке:
```
Допустим, что 192.168.5.155 - IP-адрес сервера

Настройка сервера:
sudo apt-get install nfs-kernel-server
sudo mkdir /var/tmp/nfs_share
sudo chmod 777 /var/tmp/nfs_share
sudo chown nobody:nogroup /var/tmp/nfs_share
sudo nano /etc/exports  # Здесь в конце файла нужно прописать строку: /var/tmp/nfs_share *(rw,sync,no_root_squash)
sudo exportfs -arvf
systemctl start nfs-kernel-server

Настройка клиента:
sudo apt-get install nfs-common
sudo mkdir ~/nfs_client_side
showmount -e 192.168.5.155  # Должно отобразиться "/var/tmp/nfs_share *"
sudo mount -t nfs 192.168.5.155:/var/tmp/nfs_share ~/nfs_client_side
```

# Критерии приёмки

- [x] Обновлена структура файлов для тестов
- [x] Добавлены тесты для всех операций с NFS-хранилищами
- [x] Покрыты все основные ошибки и edge cases
- [x] Тесты используют реальную инфраструктуру (NFS)
- [x] Добавлены фикстуры для NFS-ресурсов
- [x] Обновлена [документация](https://github.com/Aerodisk/openvair-docs/pull/18) тестов Storage
- [x] Обеспечено стабильное прохождение тестов:
<img width="351" height="60" alt="image" src="https://github.com/user-attachments/assets/838c39d1-dad1-4104-883a-e886bcd03437" />


# Связанные задачи

- Issue #252

Также исправлено:
- Issue #269 (исправлено)

dr-wh0 committed:
Add nfs storage API tests

dr-wh0 committed:
Fix nfs_storage deletion in fixture

dr-wh0 committed:
Refactor fixtures location in storage nfs tests

dr-wh0 committed:
Fix NfsVolume - add clone() method

dr-wh0 committed:
Add unauthorized creation test in NFS Storage tests

dr-wh0 committed:
Refactor storage NFS tests names

dr-wh0 committed:
Merge branch 'dev' into tests/storage_nfs

dr-wh0 committed:
Refactor test files names: test_retrive/test_get -> test_retrieve

dr-wh0 committed:
Merge branch 'tests/storage' into tests/storage_nfs

dr-wh0 committed:
Merge branch 'dev' into tests/storage_nfs

dr-wh0 committed:
Merge branch 'dev' into tests/storage_nfs

dr-wh0 committed:
Fix ruff and imports after merge

dr-wh0 committed:
Merge branch 'dev' into tests/storage_nfs

dr-wh0 committed:
Fix partition tests in storage API tests

- Partitions tests moved to different directory
- Added support of normal disk (not partition) as TEST_STORAGE_PATH

VladimirTrue reviewed (commented):
По логике проверок в тестах, вопросов нет. 
Но нахожу важным попробовать разобраться с фикстурами для localfs и nfs и вытекающих из этого потенциальных правок.

dr-wh0 commented on openvair/conftest.py:
По параметризации - есть несколько способов. 

1. Самый банальный - это сделать действительно параметризацию, но тогда перед каждым тестом NFS придётся дописывать строчку по типу:
```python
@pytest.mark.parametrize('storage', ['nfs'], indirect=True)
``` 
Что будет выглядит не очень, как по-мне, но это именно параметризация.

2. Менее банальный вариант - в сами фикстуры прописать, например, проверку имени файла (т.к. у нас все nfs тесты вынесены в файлы с "nfs" в названии), т.е. по сути `if "nfs" in test_path.lower()`. Это не параметризация по сути и корректность под вопросом, но зато меньше кода. 

Какой вариант тут лучше использовать?

dr-wh0 committed:
Refactor: Fix review
- Delete cleanup_storages usage because of autouse
- Change default storage_nfs_path to default=Path()

dr-wh0 commented on openvair/modules/storage/tests/api/nfs/test_create_nfs.py:
```diff
@@ -0,0 +1,205 @@
+# ruff: noqa: ARG001 because of fixtures using
```
Убрал использование "cleanup_storages: None", как в других тестах, убрал подавление

dr-wh0 committed:
Refactor storage fixture: add storage_type parametrize

tihon49 committed:
Merge branch 'dev' into tests/storage_nfs
